### PR TITLE
feat(OpenAPI): Define error schemas as components and reference them in API paths

### PR DESCRIPTION
**Pull request by abdulhaq-e**
### Pull Request Checklist

[//]: # "Please review the [Litestar contribution guidelines](https://github.com/litestar-org/litestar/blob/main/CONTRIBUTING.rst) for this repository."

- [ ] New code has 100% test coverage
- [ ] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [ ] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR

[//]: # "By submitting this issue, you agree to:"
[//]: # "- follow Litestar's [Code of Conduct](https://github.com/litestar-org/.github/blob/main/CODE_OF_CONDUCT.md)"
[//]: # "- follow Litestar's [Contribution Guidelines](https://litestar.dev/community/contribution-guide)"

### Description

Schemas for objects that were returned for `status_code=400` (and other custom exceptions) were inlined for each API path in the OpenAPI schema.  While this caused an issue for one openapi generator [library](https://github.com/openapi-generators/openapi-python-client), it is generally a good idea to define schemas as components and reference them only in each API path.

This PR implements this feature.

[//]: # "Please describe your pull request for new release changelog purposes"
[//]: # "Example: 'This pr adds the capability to use server-sent events, and documents it in the usage docs'"

### Close Issue(s)

[//]: # "Please add in issue numbers this pull request will close, if applicable"
[//]: # "Examples: Fixes #4321 or Closes #1234"

### CODE REVIEW

**Comment by abdulhaq-e** [comment]
Converted this to draft, I need to test this with an actual output, the tests pass but I need to see an actual schema first.

**Comment by abdulhaq-e** [comment]
Screen Shot 2023-07-13 at 7 53 24 PM

All good, this can be reviewed.

**Comment by abdulhaq-e** [comment]
I'll fix the failing tests tomorrow.

**Comment by Goldziher** [comment]
@abdulhaq-e ?

**Comment by abdulhaq-e** [comment]
> @abdulhaq-e ?

I'll get back to this asap. Only the tests need fixing. Sorry for keeping it open like this.

File: litestar/_openapi/responses.py [hunk 1]
```diff
@@ -204,9 +207,12 @@ def create_error_responses(exceptions: list[type[HTTPException]]) -> Iterator[tu
             schema = Schema(one_of=exceptions_schemas)
         else:
             schema = exceptions_schemas[0]
+
+        _schema = schema_creator.process_schema_result(schema)
```
Comment (Goldziher) [inline litestar/_openapi/responses.py:211]: Y the underscore?

Comment (abdulhaq-e) [reply litestar/_openapi/responses.py:211]: mypy/pyright complaining i think, i'll check again.

**Pull request closed without merging**